src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -67,6 +67,20 @@ trait GenTraversableOnce[+A] extends Any {
     */
   def foreach[U](f: A => U): Unit
 
+  /** Tests whether this $coll is known to have a finite size.
+    *  All strict collections are known to have finite size. For a non-strict
+    *  collection such as `Stream`, the predicate returns `'''true'''` if all
+    *  elements have been computed. It returns `'''false'''` if the stream is
+    *  not yet evaluated to the end. Non-empty Iterators usually return
+    *  `'''false'''` even if they were created from a collection with a known
+    *  finite size.
+    *
+    *  Note: many collection methods will not work on collections of infinite sizes
+    *  (but they do not require this method to return `'''true'''`).
```

Comment:
It's not about knowing the size but about knowing whether the collection has a finite size. And in particular, it *is* about calling this specific method. I can't come up with an equally short version that makes it clearer. How about this longer explanation instead?

> Note: many collection methods will not work on collections of infinite sizes. The typical failure mode is an infinite loop. These methods always attempt a traversal without checking first that `hasDefiniteSize` returns `true`. The latter is sufficient for a terminating traversal but it is not required.